homeassistant/components/matter/lock.py

```diff
@@ -89,21 +89,15 @@ async def send_device_command(
 
     async def async_lock(self, **kwargs: Any) -> None:
         """Lock the lock with pin if needed."""
-        code: str = kwargs.get(
-            ATTR_CODE,
-            self._lock_option_default_code,
-        )
+        code: Optional[str] = kwargs.get(ATTR_CODE)
```

Comment:
please use new style type hints
```suggestion
        code: str | None = kwargs.get(ATTR_CODE)
```

Reply:
Done. (Shouldn't the linter take care of that?)